compiler/grainformat/reformat.re

```diff
@@ -3018,183 +3843,158 @@ let data_print =
           | Nonexported => Doc.nil
           | Exported => Doc.text("export ")
           },
-          print_data(decl, original_source),
+          print_data(~original_source, ~comments, decl),
         ]);
       },
       datas,
     ),
   );
 };
-let import_print = (imp: Parsetree.import_declaration) => {
+let import_print =
+    (
+      ~comments: list(Parsetree.comment),
+      ~original_source: array(string),
+      imp: Parsetree.import_declaration,
+    ) => {
   let vals =
     List.map(
       (v: Parsetree.import_value) => {
         switch (v) {
         | PImportModule(identloc) => print_ident(identloc.txt)
         | PImportAllExcept(identlocs) =>
-          let numVals = List.length(identlocs);
-
           Doc.concat([
             Doc.text("*"),
-            if (List.length(identlocs) > 0) {
+            switch (identlocs) {
+            | [] => Doc.nil
+            | [first, ...rem] =>
+              let get_loc = (identloc: Location.loc(Identifier.t)) => {
+                identloc.loc;
+              };
+
+              let print_item =
+                  (identloc: Location.loc(Identifier.t), comments) => {
+                print_ident(identloc.txt);
+              };
+
+              let after_brace_comments =
+                Comment_utils.get_after_brace_comments(
+                  imp.pimp_loc,
+                  comments,
+                  None,
+                );
+              let cleaned_comments =
+                remove_used_comments(
+                  ~all_comments=comments,
+                  after_brace_comments,
+                );
+
+              let exceptions =
+                block_item_iterator_line(
+                  ~previous=Block(imp.pimp_loc),
+                  ~get_loc,
+                  ~print_item,
+                  ~get_attribute_text=_ => Doc.nil,
```

Comment:
I wonder if `~get_attribute_text` should be an optional argument, so we don't need these kind of functions

Reply:
I removed them where we were passing in this Doc.nil as they were unnecessary.  The get_attribute_text is always used for block_item_iterator so I've kept it (as we pass in expressions or top level statements)